tests/libstorage/storageclass.go

```diff
@@ -121,17 +118,27 @@ func CheckNoProvisionerStorageClassPVs(storageClassName string, numExpectedPVs i
 	// Verify we have at least `numExpectedPVs` available file system PVs
 	pvList, err := virtClient.CoreV1().PersistentVolumes().List(context.TODO(), metav1.ListOptions{})
 	Expect(err).ToNot(HaveOccurred())
+
+	if countLocalStoragePV(pvList, storageClassName) < numExpectedPVs {
+		Skip("Not enough available filesystem local storage PVs available, expected: %d", numExpectedPVs)
+	}
+}
+
+func countLocalStoragePV(pvList *k8sv1.PersistentVolumeList, storageClassName string) int {
 	count := 0
 	for _, pv := range pvList.Items {
-		if pv.Spec.StorageClassName != storageClassName || pv.Spec.NodeAffinity == nil || pv.Spec.NodeAffinity.Required == nil || len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms) == 0 || (pv.Spec.VolumeMode != nil && *pv.Spec.VolumeMode == k8sv1.PersistentVolumeBlock) {
-			// Not a local volume filesystem PV
-			continue
-		}
-		if pv.Spec.ClaimRef == nil {
```

Comment:
Well, this doesn't say "I am not local", @brybacki keep me honest here...

Reply:
@xpivarc 
very good eye, this means a PV is not yet claimed, and is still free to be bound, 

and now I think we can writte it even better:
1. name the function 
```
func countLocalStoragePVAvailableForUse(pvList *k8sv1.PersistentVolumeList, storageClassName string) int {
count := 0
	for _, pv := range pvList.Items {
        isLocalPV := pv.Spec.StorageClassName == storageClassName &&
		    pv.Spec.NodeAffinity != nil &&
	    	pv.Spec.NodeAffinity.Required != nil &&
    		len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms) > 0 &&
		    (pv.Spec.VolumeMode == nil || *pv.Spec.VolumeMode != k8sv1.PersistentVolumeBlock)
        isNotClaimed := pv.Spec.ClaimRef
		
		if isLocalPV && isNotClaimed {
			count++
		}
	}
	return count
}
 
```

Reply:
I like what @brybacki wrote down as it reads as something that I expect it is doing. You can use functions rather than the expression evaluation(`isLocalPV`).